File: frontend/src/utils/textConverter.ts
/**
 * 文本转换工具
 * 
 * 提供繁体中文转简体中文等功能
 */

import { toSimplified, customT2SPhrases } from 'chinese-simple2traditional';
import { setupEnhance } from 'chinese-simple2traditional/enhance';

// 初始化短语库（只初始化一次）
let enhanceInitialized = false;

function initializeEnhance() {
  if (!enhanceInitialized) {
    setupEnhance();
    
    // 添加自定义短语修复已知的转换错误
    // 注意：customT2SPhrases 的参数是 [繁体短语, 简体短语] 的数组
    customT2SPhrases([
      // 修复"退休"相关词汇
      ['退休', '退休'],  // 修复：退休 -> 退休（而不是煺休）
      ['退休人員', '退休人员'],  // 修复：退休人员
      ['退休金', '退休金'],  // 修复：退休金
      ['退休工資', '退休工资'],  // 修复：退休工资
      // 修复"怎么"相关词汇
      ['怎麽', '怎么'],  // 修复：怎麽 -> 怎么
      ['怎麼樣', '怎么样'],  // 修复：怎么样 -> 怎么样
      ['怎麽樣', '怎么样'],  // 修复：怎麽样 -> 怎么样（备用）
    ]);
    
    enhanceInitialized = true;
  }
}

// 初始化短语库
initializeEnhance();

/**
 * 将繁体中文转换为简体中文
 * 
 * @param text - 输入文本（可能是繁体或简体）
 * @returns 转换后的简体中文文本
 * 
 * @example
 * ```ts
 * toSimplifiedChinese('繁體中文') // '繁体中文'
 * ```
 */
export function toSimplifiedChinese(text: string): string {
  if (!text) {
    return text;
  }

  try {
    // 使用增强模式（第二个参数传 true）以获得更精确的转换
    // 增强模式会使用短语库，避免单个字符转换导致的错误
    const converted = toSimplified(text, true);
    return converted;
  } catch (error) {
    // 转换失败时返回原文本
    console.warn('Failed to convert text to simplified Chinese:', error);
    return text;
  }
}

/**
 * 检测文本是否包含繁体中文
 * 
 * @param text - 输入文本
 * @returns 如果包含繁体中文返回true，否则返回false
 */
export function isTraditionalChinese(text: string): boolean {
  if (!text) {
    return false;
  }

  try {
    // 如果转换后文本有变化，说明包含繁体
    const converted = toSimplified(text);
    return converted !== text;
  } catch {
    return false;
  }
}

File: backend/app/utils/text_converter.py
"""
文本转换工具

提供繁体中文转简体中文等功能
"""

# 标准库
from typing import Optional

# 本地库
from app.utils.logger import logger

try:
    # 尝试导入 zhconv（如果已安装）
    # 类型存根文件位于 stubs/zhconv/__init__.pyi
    import zhconv
    ZHCONV_AVAILABLE = True
except ImportError:
    ZHCONV_AVAILABLE = False
    logger.warning("zhconv not installed, traditional to simplified conversion will be disabled")


def to_simplified(text: str) -> str:
    """将繁体中文转换为简体中文
    
    Args:
        text: 输入文本（可能是繁体或简体）
    
    Returns:
        str: 转换后的简体中文文本
    
    Note:
        如果 zhconv 未安装，将返回原文本
    """
    if not text:
        return text
    
    if not ZHCONV_AVAILABLE:
        logger.debug("zhconv not available, skipping conversion")
        return text
    
    try:
        # zhconv.convert 会自动检测并转换繁体到简体
        # 如果文本已经是简体，不会改变
        converted = zhconv.convert(text, "zh-cn")
        
        # 记录转换情况（仅在确实有变化时记录）
        if converted != text:
            logger.debug(
                "Converted traditional Chinese to simplified",
                extra={
                    "original_length": len(text),
                    "converted_length": len(converted),
                    "original_preview": text[:50] if len(text) > 50 else text,
                    "converted_preview": converted[:50] if len(converted) > 50 else converted
                }
            )
        
        return converted
    except Exception as e:
        logger.warning(
            f"Failed to convert text to simplified Chinese: {e}",
            exc_info=True
        )
        # 转换失败时返回原文本
        return text


def is_traditional_chinese(text: str) -> bool:
    """检测文本是否包含繁体中文
    
    Args:
        text: 输入文本
    
    Returns:
        bool: 如果包含繁体中文返回True，否则返回False
    
    Note:
        如果 zhconv 未安装，将返回False
    """
    if not text or not ZHCONV_AVAILABLE:
        return False
    
    try:
        # 如果转换后文本有变化，说明包含繁体
        converted = zhconv.convert(text, "zh-cn")
        return converted != text
    except Exception:
        return False

